lib/sgtn-client/api/translation.rb

```diff
@@ -12,24 +12,16 @@ module Core
   module Translation
     def self.getString(component, key, locale)
       SgtnClient.logger.debug "[Translation.getString]component: #{component}, key: #{key}, locale: #{locale}"
-      str = getTranslation(component, key, locale)
-      if str.nil? && !LocaleUtil.is_source_locale(locale)
-        str = getTranslation(component, key, LocaleUtil.get_source_locale)
-      end
-      str
+      getTranslation(component, key, locale)
```

Comment:
Isn't `getTranslation` possible return `nil`?

Reply:
It won't anymore because of the following line in fetch:` return source_bundle, nil, LocaleUtil.get_source_locale if translation_bundle.nil? `

This means we always write something in cache. If translation fetch for target locale fails, we write a cache_item that has a reference to the source bundle (because it has fallen back to the source bundle). This avoids the many requests for a translation bundle that is not yet ready, or for when fetch fails for any other reason (e.g. VIP service is down).

Take note that we won't duplicate the source bundle in memory; it only saves a reference to the already existing source bundle in memory.